PXL contract 수정

charlskim opened the pull request:
- Openzeppelin 최신 커밋 버전으로 수정
- Token contract에서 sale 참여자 별로 락업 기간을 확인하도록 수정, 토큰은 sale 참여 시 바로 전송
- 한 지갑당 sale은 하나만 참여 가능하며 여러 sale에 참여할 경우 지갑을 달리하여 참여하여야 함
  (공지 필요: private, presale 1차, presale 2차의 락업 기간이 다를 경우 같은 주소의 참여자가 참여 했을 시 문제가 발생함으로 각 sale마다 다른 지갑으로 참여해야함)

charlskim committed:
PXL contract 수정
- Openzeppelin 최신 커밋 버전으로 수정

charlskim committed:
compile 에러 수정

charlskim committed:
Openzeppelin math 최신 커밋으로 코드 수정

charlskim committed:
개인 Lockup 설정 코드 추가

Commit:
컴파일 에러 수정

Commit:
Pxl 테스트 코드 추가

charlskim committed:
토큰 판매 정책 변경으로 인한 미 사용 컨트렉트 주석 추가

charlskim committed:
Merge branch 'feature/PXL' of https://github.com/piction-protocol/piction-ico into feature/PXL

Commit:
pxl 테스트 코드 TransferFrom 추가

Commit:
Merge branch 'feature/PXL' of https://github.com/piction-protocol/piction-ico into feature/PXL

charlskim committed:
Pxl 컨트랙트 생성자 변경 : ledger 계정이 owner가 되도록 변경

charlskim committed:
Merge branch 'feature/PXL' of https://github.com/piction-protocol/piction-ico into feature/PXL

jhseo commented on contracts/token/PXL.sol:
```diff
@@ -135,6 +134,10 @@ contract PXL is StandardToken, CustomToken, ExtendsOwnable {
     }
 
+    function getLockup(address _account) public view returns (uint256) {
+        return lockup[_account];
```
이미 lockup에 mapping된 account를 체크하지 않아도 될까요?

jhseo commented on contracts/token/PXL.sol:
```diff
@@ -81,14 +82,19 @@ contract PXL is StandardToken, CustomToken, ExtendsOwnable {
+    }
+
+    function transferLockup(address _to, uint256 _value, uint256 _days) public onlyOwner returns (bool) {
+        lockup[_to] = _days * 1 days;
```
이미 lockup에 mapping된 account인지 체크하지 않아도 될까요?

charlskim replied:
네. 필요한 코드네요!
감사합니다~ :)

charlskim committed:
락업 확인 코드 추가

skkwon80 commented:
ㅇㅇㅇ

charlskim commented:
네. setter 함수 추가하겠습니다.

charlskim committed:
개인 락업 기간 변경 함수 추가

skkwon80 commented:
락업되는 토큰에 대해 이벤트를 남겨도 좋을거 같습니다.

charlskim committed:
락업 이벤트 추가

charlskim committed:
한글 주석 추가

charlskim committed:
함수명 변경(transferLockup -> transferAndLockup)

charlskim committed:
생성자 함수 제거 : 생성자에서 transferOwnership을 사용하지 못함

Commit:
PXL 코드 수정으로 인한 테스트 코드 변경

Commit:
setLockup 테스트 코드 추가

charlskim committed:
최종 판매 정책으로 토큰 컨트랙트 수정

charlskim committed:
Merge branch 'feature/PXL' of https://github.com/piction-protocol/piction-ico into feature/PXL

dmbyun commented on contracts/token/PXL.sol:
```diff
@@ -16,79 +16,68 @@ import "contracts/utils/ExtendsOwnable.sol";
-     * @return A boolean that indicates if transfer was successful.
+     * @notice 토큰 전송이 불가능 할 경우 전송 실패
+     * @param _from 토큰을 가지고 있는 지갑 주소
+     * @param _to 대리 전송 권한을 부여할 지갑 주소
```
이부분은 토큰 대리 권한이 아닌 '토큰을 받을 주소'가 맞을 것 같습니다.
권한은 Approve 에서 얻은 msg.sender가 가지고 있는게 맞는것 같습니다.

dmbyun replied:
제가 테스트 코드 수정하며 고칠께요 ^^

Commit:
pxl 테스트 코드 수정

charlskim committed:
Openzeppelin-Solidity 최신 버전으로 수정

charlskim committed:
Openzeppelin 최신 코드로 수정
- ERC20.sol 파일로 통합(StandardToken.sol 삭제)
- SafeMath.sol import문 제거(ERC20.sol에 포함)

charlskim committed:
ERC20 최신 코드 적용(중복 import 제거)
